runtime-testsuite/test/org/antlr/v4/test/runtime/BaseRuntimeTest.java

```diff
@@ -42,18 +44,10 @@
  *  @since 4.6.
  */
 public abstract class BaseRuntimeTest {
-
-	public final static String[] Targets = {
-		"Cpp",
-		"CSharp",
-		"Dart",
-		"Go",
-		"Java",
-		"Node",
-		"PHP",
-		"Python2", "Python3",
-		"Swift"
-	};
+	final static Set<String> sections = new HashSet<>(Arrays.asList(
+		"notes", "type", "grammar", "slaveGrammar", "start", "input", "output", "errors", "flags", "skip"
+	));
+	final static Pattern linePattern = Pattern.compile("([^\\r\\n]*)(\r?\n)");
```

Comment:
to simplify how about a-zA-Z instead of the `^` based one?

Reply:
It's not correct, because we have to read files line by line separated by line separators. File content contains different words and chars.

Reply:
oh so you see why i don't like regex. i quickly looked and saw `[...]` as matching the section names. haha.  why not split on `\r?\n` then?

Reply:
Because in some tests line separators matter. Specifically, in tests in this PR: `LineSeparator_LF.txt eol=lf, LineSeparator_CRLF.txt`. Also, in general, input changing is not good.

Reply:
Yeah,I was wondering about that... What about `.*?\r?\n` for the regex?

BTW, do we assume that the same line Terminator is used throughout the entire test descriptor file? I think we should.

Reply:
> Yeah,I was wondering about that... What about .*?\r?\n for the regex?

Yes, it's possible to use `(.*?)(\r?\n)` (with groups) but I don't see a big difference. Also, I try to get rid of greedy quantifiers if possible.

> BTW, do we assume that the same line Terminator is used throughout the entire test descriptor file? I think we should.

Line terminator may differ even within one section (it's relevant for input). For other sections line terminator does not matter (and it's being replaced by `\n` because ANTLR uses `\n` for output).

Reply:
Ah, yes, you're right about input/output. Hm..ok, this is close enough. let's go for it.